### ISSUE TITLE
VSCode very laggy and dependent on window size

### ISSUE THREAD

**Issue by ewantowie** [microsoft/vscode#34940]
- VSCode Version: Code 1.15.1 (41abd21afdf7424c89319ee7cb0445cc6f376959, 2017-08-16T17:16:27.618Z)
- OS Version: Linux x64 3.10.0-514.6.1.el7.x86_64
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
bracket-pair-colorizer|Coe|0.10.9
vscode-eclipse-keybindings|alp|0.9.6
Doxygen|bbe|1.0.0
cpptools|ms-|0.12.4
funclist|qrt|7.0.0

(1 theme extensions excluded)

---

Steps to Reproduce:

1. Open any file, maximise window to fill desktop.
2. Feel keyboard lag and slowly increase frustration levels ;)

Reproduces without extensions: Yes

Basically, if I have the vscode window open in full screen the gui response lag for vscode is really quite bad. If I make the window smaller the lag time is reduced, and if the window is postage stamp sized the lag is useable.

Please ask for any specific info you may need to help resolve this issue.

**Comment by alexandrudima** [comment]
This might be a dup of #9432

**Comment by ewantowie** [comment]
I have just upgraded to 1.16.1 to see if the issue was resolved, but it still seems to be evident. Using a smaller window still reduces the UI lag.

For reference my desktop resolution is 3834x2052 pixels

**Comment by bpasero** [comment]
@ewantowie any change with 1.17 release?

**Comment by ewantowie** [comment]
It has improved, although there is still a lag difference between maximised and small window sizes.

I don't know if it is connected, but I have also discovered that I cannot get another window to show in front of VSCode. I have to minimise VSCode in order to use any other window, very odd!

**Comment by bpasero** [comment]
@ewantowie that is weird indeed, can you file a separate issue.

**Comment by ewantowie** [comment]
@bpasero I have created a separate issue #36218 for the window maximise problem.

**Comment by wolfhoundjesse** [comment]
I've tried the suggestions above, but I experience lag, too. Running Code with Optirun doesn't help much, either. I'm scaling with xrandr (2x2) because my laptop is 4k and my external monitors are not. My current environment:

- Arch Linux 4.13.9-1-ARCH
- Dell Precision 5520
- 8 Cores, 32GB RAM (neither seems particularly taxed while running code)

**Comment by bpasero** [comment]
We are building exploration builds that use a much newer version of our UI framework (Electron version 6.0.x). I wonder if this issue reproduces with one of these builds, could you try? Download:
* macOS: [Download](https://az764295.vo.msecnd.net/exploration/960208fdf9aa06e6892aa14c9c84ddd607fbeb2f/VSCode-darwin-exploration.zip)
* Linux (DEB): [Download](https://az764295.vo.msecnd.net/exploration/960208fdf9aa06e6892aa14c9c84ddd607fbeb2f/code-exploration_1.37.0-1564556412_amd64.deb)
* Linux (RPM): [Download](https://az764295.vo.msecnd.net/exploration/960208fdf9aa06e6892aa14c9c84ddd607fbeb2f/code-exploration-1.37.0-1564556601.el7.x86_64.rpm)
* Windows: [Download](https://az764295.vo.msecnd.net/exploration/960208fdf9aa06e6892aa14c9c84ddd607fbeb2f/VSCodeUserSetup-x64-1.37.0-exploration.exe)

**Comment by ewantowie** [comment]
Could you prepare a tarball for Linux as well?

**Comment by bpasero** [comment]
Yes: https://az764295.vo.msecnd.net/exploration/a715df3459718a501686bd03414f69929bf6e9a2/code-exploration-1568356741.tar.gz

**Comment by ewantowie** [comment]
Thanks! I tried the tarball on my workstation but unfortunately I can't run the binary as it dumps the following error:

./VSCode-linux-x64-exploration/code-exploration 
[185008:1011/095219.694227:FATAL:setuid_sandbox_host.cc(157)] The SUID sandbox helper binary was found, but is not configured correctly. Rather than run without sandboxing I'm aborting now. You need to make sure that /remote/gb12tcad3/ewan-nis/Apps/VSCode-linux-x64-exploration/chrome-sandbox is owned by root and has mode 4755.
Trace/breakpoint trap

I am not comfortable in making executables root owned for what is a local installation. Can you implement a work around so I can test this for you?

**Comment by bpasero** [comment]
@ewantowie you can run with "`--no-sandbox`"

**Comment by ewantowie** [comment]
Ok, doing a very simple test (scrolling a long text file) I can see no real difference in performance between this version and 1.38.1 with a maximised window on a 4K dekstop. I am running these on a CentOS 7.3 box.

Certainly, the full screen performance in 1.38.1 is much better than in 1.15.1 for scrolling a text file.

Is there any particular test or metric you would like to see?

**Comment by ewantowie** [comment]
@bpasero 

I can certainly repeat the issue in Electron, but I cannot get the web based client to show anything. The newest version of Chromium I have available is 14.0, so could this be the issue?

**Comment by bpasero** [comment]
@ewantowie you should see a browser open and show something like this:

![image](https://user-images.githubusercontent.com/900690/67287355-3f635380-f4db-11e9-85fa-329e5b722585.png)

Do you get a browser tab to open? If the page is empty, do you see anything in the developer console?

**Comment by ewantowie** [comment]
I get a browser tab with an empty page, and the developer console has a lsit of syntax errors.

![SCREEN~2](https://user-images.githubusercontent.com/32168905/67383462-18faf200-f587-11e9-8fdc-0745afd8e24b.PNG)

**Comment by bpasero** [comment]
Strange, is that Chrome browser latest? I think your browser does not support running VSCode web.

**Comment by ewantowie** [comment]
Apologies, didn't get a notification that you had responded here. The version of Chrome is probably quite old, we predominantly use Firefox on our linux boxes.

I mentioned above that we are running Chromium 14.0.785.0 (Developer Build 87778 Linux).

**Comment by ewantowie** [comment]
@bpasero The bot has closed this issue, not sure it really wants to :)

**Comment by bpasero** [comment]
This is probably the same cause as https://bugs.chromium.org/p/chromium/issues/detail?id=1016329

**Comment by ewantowie** [comment]
If it's any help, we don't run any GPUs on the machine I use as they are all VMs running off a single node. I believe all the graphics for Xorg is software based.

**Comment by bpasero** [comment]
This is not actionable from our end, closing.
